homeassistant/components/nissan_leaf/__init__.py

```diff
@@ -396,79 +375,63 @@ def _extract_start_date(battery_info):
         """Set climate control mode via Nissan servers."""
         climate_result = None
         if toggle:
-            _LOGGER.info("Requesting climate turn on for %s", self.leaf.vin)
-            request = await self.hass.async_add_job(
-                self.leaf.start_climate_control
-            )
-            for attempt in range(MAX_RESPONSE_ATTEMPTS):
-                if attempt > 0:
-                    _LOGGER.info("Climate data not in yet (%s) (%s). "
-                                 "Waiting (%s) seconds.", self.leaf.vin,
-                                 attempt, PYCARWINGS2_SLEEP)
-                    await asyncio.sleep(PYCARWINGS2_SLEEP)
-
-                climate_result = await self.hass.async_add_job(
-                    self.leaf.get_start_climate_control_result, request
-                )
-
-                if climate_result is not None:
-                    break
-
+            _LOGGER.debug("Requesting climate turn on for %s", self.leaf.vin)
+            set_function = self.leaf.start_climate_control
+            result_function = self.leaf.get_start_climate_control_result
         else:
-            _LOGGER.info("Requesting climate turn off for %s", self.leaf.vin)
-            request = await self.hass.async_add_job(
-                self.leaf.stop_climate_control
-            )
+            _LOGGER.debug("Requesting climate turn off for %s", self.leaf.vin)
+            set_function = self.leaf.stop_climate_control
+            result_function = self.leaf.get_stop_climate_control_result
 
-            for attempt in range(MAX_RESPONSE_ATTEMPTS):
-                if attempt > 0:
-                    _LOGGER.debug("Climate data not in yet. (%s) (%s). "
-                                  "Waiting %s seconds", self.leaf.vin,
-                                  attempt, PYCARWINGS2_SLEEP)
-                    await asyncio.sleep(PYCARWINGS2_SLEEP)
-
-                climate_result = await self.hass.async_add_job(
-                    self.leaf.get_stop_climate_control_result, request
-                )
+        request = await self.hass.async_add_executor_job(set_function)
+        for attempt in range(MAX_RESPONSE_ATTEMPTS):
+            if attempt > 0:
+                _LOGGER.debug("Climate data not in yet (%s) (%s). "
+                              "Waiting (%s) seconds", self.leaf.vin,
+                              attempt, PYCARWINGS2_SLEEP)
+                await asyncio.sleep(PYCARWINGS2_SLEEP)
 
-                if climate_result is not None:
-                    break
+            climate_result = await self.hass.async_add_executor_job(
+                result_function, request)
+
+            if climate_result is not None:
+                break
 
         if climate_result is not None:
             _LOGGER.debug("Climate result: %s", climate_result.__dict__)
-            async_dispatcher_send(self.hass, SIGNAL_UPDATE_LEAF)
+            dispatcher_send(self.hass, SIGNAL_UPDATE_LEAF)
```

Comment:
Revert this. We're in async context since we're in a coroutine.

Reply:
Fixed with commit https://github.com/home-assistant/home-assistant/pull/21145/commits/b961a8f38cc6ed433d85542e5110e317a11479ab